File: lab02/zad2/src/MyData.java
import java.net.InetAddress;
import java.net.UnknownHostException;
import java.text.SimpleDateFormat;
import java.util.Date;

public class MyData {

    public static void main(String[] args) {

    }
    public static void info(){

        System.out.println("Filip Strózik, 260377");
        System.out.println("Piotr Grygoruk, 260299");

        SimpleDateFormat tf = new SimpleDateFormat("dd MMM, HH:mm:ss");
        System.out.println(tf.format(new Date()));

        System.out.println(System.getProperty("java.version"));
        System.out.println(System.getProperty("user.name"));
        System.out.println(System.getProperty("os.name"));
        try {
            System.out.println(InetAddress.getLocalHost().getHostAddress());
        } catch (UnknownHostException e) {
            throw new RuntimeException(e);
        }

    }
}

File: lab04/1b/GrpcGreeter/GrpcGreeter/Services/MyData.cs
using Microsoft.Win32;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Net.Sockets;
using System.Text;
using System.Threading.Tasks;

namespace grpcGreeter.Services
{
    public class MyData
    {
        public static void Info()
        {
            Console.WriteLine("Filip Strózik, 260377");
            Console.WriteLine("Piotr Grygoruk, 260299");

            Console.WriteLine((DateTime.UtcNow).ToString("dd MMM, HH:mm:ss"));

            Console.WriteLine(Environment.Version.ToString());
            Console.WriteLine(Environment.UserName);
            Console.WriteLine(Environment.OSVersion);

            try
            {
                Console.WriteLine(GetLocalIPAddress()); 
            }
            catch (Exception e)
            {
                throw new Exception(e.Message);
            }
        }

        public static string GetLocalIPAddress()
        {
            var host = Dns.GetHostEntry(Dns.GetHostName());
            foreach (var ip in host.AddressList)
            {
                if (ip.AddressFamily == AddressFamily.InterNetwork)
                {
                    return ip.ToString();
                }
            }
            throw new Exception("No network adapters with an IPv4 address in the system!");
        }
    }
}
